include/stdexec/execution.hpp

```diff
@@ -5736,13 +5737,359 @@ namespace stdexec {
+        void operator()(__mtype<_T>, _Storage& __self, const _Storage& __from) {
+          tag_invoke(__copy_construct_t{}, __mtype<_T>{}, __self, __from);
+        }
+    };
+    inline constexpr __copy_construct_t __copy_construct{};
+
+    struct __move_construct_t {
+      template <class _Storage, class _T>
+          requires tag_invocable<__move_construct_t, __mtype<_T>, _Storage&,  _Storage&&>
+        void operator()(__mtype<_T>, _Storage& __self, __midentity<_Storage&&> __from) noexcept {
+          tag_invoke(__move_construct_t{}, __mtype<_T>{}, __self, (_Storage&&) __from);
+        }
+    };
+    inline constexpr __move_construct_t __move_construct{};
+
+    template <class _ParentVTable, class... _StorageCPOs>
+      struct __storage_vtable : _ParentVTable, __storage_vfun<_StorageCPOs>... {
+        using _ParentVTable::operator();
+        using __storage_vfun<_StorageCPOs>::operator()...;
+      };
+
+    template <class _ParentVTable, class... _StorageCPOs>
+      inline constexpr __storage_vtable<_ParentVTable, _StorageCPOs...> __null_storage_vtbl {};
+
+    template <class _ParentVTable, class... _StorageCPOs>
+      constexpr const __storage_vtable<_ParentVTable, _StorageCPOs...>*
+        __default_storage_vtable(__storage_vtable<_ParentVTable, _StorageCPOs...>*) noexcept {
+          return &__null_storage_vtbl<_ParentVTable, _StorageCPOs...>;
+        }
+
+    template <class _Storage, class _T, class _ParentVTable, class... _StorageCPOs>
+      inline constexpr __storage_vtable<_ParentVTable, _StorageCPOs...>
+        __storage_vtbl {
+          {*__create_vtable<_ParentVTable>(__mtype<_T>{})},
+          {__storage_vfun_fn<_Storage, _T>{}((_StorageCPOs*) nullptr)}...
+        };
+
+    template <class _Allocator,
+              bool _Copyable = false,
+              std::size_t _Alignment = alignof(std::max_align_t), 
+              std::size_t _InlineSize = 3*sizeof(void*)>
+    struct __basic_storage {
+      template <class _Vtable>
+        struct __storage {
+          class __t;
+        };
+    };
+
+    template <class _Allocator = std::allocator<std::byte>>
+      using __unique_storage = __basic_storage<_Allocator>;
+
+    template <class _Allocator = std::allocator<std::byte>>
+      using __copyable_storage = __basic_storage<_Allocator, true>;
+
+    template <class _Allocator, bool _Copyable, std::size_t _Alignment, std::size_t _InlineSize>
+    template <class _Vtable> 
+      class __basic_storage<_Allocator, _Copyable, _Alignment, _InlineSize>::__storage<_Vtable>::__t {
+        static constexpr std::size_t __buffer_size = std::max(_InlineSize, sizeof(void*));
+        static constexpr std::size_t __alignment = std::max(_Alignment, alignof(void*));
+        using __with_copy = __copy_construct_t(void(*)(const __t&));
+        using __with_move = __move_construct_t(void(*)(__t&&) noexcept);
+        using __with_delete = __delete_t(void(*)() noexcept);
```

Comment:
Incidentally, these can be simply:

```suggestion
        using __with_copy = __copy_construct_t(void(const __t&));
        using __with_move = __move_construct_t(void(__t&&) noexcept);
        using __with_delete = __delete_t(void() noexcept);
```

C++ is weird like that.